File: ios/bridge/ExpoBridgeRecommendations.swift
import Foundation

@available(iOS 16.0, *)
enum ExpoBridgeRecommendations {
  static func getRecommendations(service: RecommendationsService, ids: [String]?) async throws -> [String: Any] {
    let recommendations = try await service.getRecommendations(ids: ids)
    return BridgeResponses.recommendations(recommendations)
  }

  static func getReplay(service: RecommendationsService, year: Int?) async throws -> [String: Any] {
    let summaries = try await service.getReplay(year: year)
    return BridgeResponses.replaySummaries(summaries)
  }
}

File: android/src/main/java/expo/modules/applemusic/bridge/ExpoBridgeRecommendations.kt
package expo.modules.applemusic.bridge

import expo.modules.applemusic.AndroidRecommendationsService
import expo.modules.applemusic.BridgeResponses
import expo.modules.kotlin.functions.Coroutine
import expo.modules.kotlin.modules.ModuleDefinition

internal fun ModuleDefinition.registerRecommendationsBridge(
  recommendationsService: () -> AndroidRecommendationsService,
) {
  AsyncFunction("getRecommendations") Coroutine { ids: List<String>? ->
    BridgeResponses.recommendations(recommendationsService().getRecommendations(ids))
  }

  AsyncFunction("getReplay") Coroutine { year: Int? ->
    BridgeResponses.replaySummaries(recommendationsService().getReplay(year))
  }
}
